sound/hda/intel-dsp-config.c

```diff
@@ -557,9 +557,32 @@ static const struct config_entry *snd_intel_dsp_find_config
 		if (table->codec_hid) {
 			int i;
 
-			for (i = 0; i < table->codec_hid->num_codecs; i++)
-				if (acpi_dev_present(table->codec_hid->codecs[i], NULL, -1))
+			for (i = 0; i < table->codec_hid->num_codecs; i++) {
+				struct nhlt_acpi_table *nhlt;
+				bool ssp_found = false;
+
+				if (!acpi_dev_present(table->codec_hid->codecs[i], NULL, -1))
+					continue;
+
+				nhlt = intel_nhlt_init(&pci->dev);
```

Comment:
If num_codecs is 4 then we do 4x  intel_nhlt_init() / intel_nhlt_free() ?
Why not do that around the for loop once?
That would be even done ARRAY_SIZE(config_table) times, so move it even further?

Reply:
OK, it is just strange to see _init() / _free() done in a loop for something which is static for the whole function.

Reply:
yeah the flow is a bit weird indeed. I just didn't want to make things too complicated.